File: 876-middle-of-the-linked-list/876-middle-of-the-linked-list.cpp
class Solution {
public:
    ListNode* middleNode(ListNode* head) {
        ListNode* ans=head;
        // ans->next=head;
        int count=0;
        while(head)
        {
            // ans=ans->next;
            head=head->next;
            if(head==NULL)
                break;
            head=head->next;
            ans=ans->next;
            
        }
        return ans;
        
    }
};

File: 876-middle-of-the-linked-list/876-middle-of-the-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode middleNode(ListNode head) {
        ListNode ans=head;
        while(head!=null){
            head=head.next;
            if(head==null){
                break;
            }
            head=head.next;
            ans=ans.next;
        }
        return ans;
    }
}